### PR TITLE
Implement parity_versionInfo & parity_setChain on LC; fix parity_setChain

### PR DESCRIPTION
**Pull request by axelchalon**
- ~~Implement eth_subscribe('syncing') RPC for full node & light node (closes https://github.com/paritytech/parity-ethereum/issues/10080); same output format as eth_syncing rpc~~: we'll use the implentation of this other PR https://github.com/paritytech/parity-ethereum/pull/10311 instead
- parity_setChain RPC: return an error if failed (instead of `true`)
- Light client: implement set_exit_handler & parity_setChain RPC (needed for Fether)
- Light client: implement parity_versionInfo RPC (needed for Fether)

 cc @amaurymartiny

### CODE REVIEW

**Comment by axelchalon** [comment]
There's another implementation of eth_subscribe('syncing') in this PR: https://github.com/paritytech/parity-ethereum/pull/10311

**Comment by axelchalon** [comment]
FWIW if preferred I can revert the `Implement eth_subscribe('syncing')` commit if we decide to use the implementation in the other PR https://github.com/paritytech/parity-ethereum/pull/10311

**Comment by Tbaut** [comment]
adding the beta label as we need if for Fether :)

File: parity/run.rs [hunk 1]
```diff
@@ -379,7 +383,7 @@ fn execute_light_impl(cmd: RunCmd, logger: Arc<RotatingLogger>) -> Result<Runnin
 
 fn execute_impl<Cr, Rr>(cmd: RunCmd, logger: Arc<RotatingLogger>, on_client_rq: Cr,
 						on_updater_rq: Rr) -> Result<RunningClient, String>
-	where Cr: Fn(String) + 'static + Send,
+	where Cr: Fn(String) -> Result<(), ()> + 'static + Send,
```
Comment (niklasad1) [inline parity/run.rs]: Is this change needed? 

It breaks `parity-clib` compatibility

File: parity/run.rs [hunk 2]
```diff
@@ -165,7 +165,9 @@ impl ::local_store::NodeInfo for FullNodeInfo {
 type LightClient = ::light::client::Client<::light_helpers::EpochFetch>;
 
 // helper for light execution.
-fn execute_light_impl(cmd: RunCmd, logger: Arc<RotatingLogger>) -> Result<RunningClient, String> {
+fn execute_light_impl<Cr>(cmd: RunCmd, logger: Arc<RotatingLogger>, on_client_rq: Cr) -> Result<RunningClient, String>
+	where Cr: Fn(String) -> Result<(), ()> + 'static + Send
```
Comment (niklasad1) [inline parity/run.rs]: Is this change needed? 

It breaks `parity-clib` compatibility

File: rpc/src/v1/impls/parity_set.rs [hunk 3]
```diff
@@ -240,7 +239,7 @@ impl<C, M, U, F> ParitySet for ParitySetClient<C, M, U, F> where
 		let hash = hash.into();
 
 		Ok(self.miner.remove_transaction(&hash)
-		   .map(|t| Transaction::from_pending(t.pending().clone()))
+			 .map(|t| Transaction::from_pending(t.pending().clone()))
```
Comment (niklasad1) [inline rpc/src/v1/impls/parity_set.rs]: remove space please :P

File: parity/run.rs (hunk 1 above)
Comment (axelchalon) [reply parity/run.rs]: OK, it'd be nice to have the RPC response reflect whether or not the restart succeeded (although not necessary) but changing the return value of the restart client callback to an integer doesn't seem very clean so I'll change back the return value of the callback for now! :+1:
